src/Model/APContact.php

```diff
@@ -189,17 +189,22 @@ public static function getByURL(string $url, $update = null): array
 		if (empty($data)) {
 			$local_owner = [];
 
-			$curlResult = HTTPSignature::fetchRaw($url);
-			$failed = empty($curlResult) || empty($curlResult->getBody()) ||
-				(!$curlResult->isSuccess() && ($curlResult->getReturnCode() != 410));
-
-			if (!$failed) {
-				$data = json_decode($curlResult->getBody(), true);
-				$failed = empty($data) || !is_array($data);
-			}
+			try {
+				$curlResult = HTTPSignature::fetchRaw($url);
+				$failed = empty($curlResult) || empty($curlResult->getBody()) ||
+					(!$curlResult->isSuccess() && ($curlResult->getReturnCode() != 410));
+	
+				if (!$failed) {
+					$data = json_decode($curlResult->getBody(), true);
+					$failed = empty($data) || !is_array($data);
+				}
 
-			if (!$failed && ($curlResult->getReturnCode() == 410)) {
-				$data = ['@context' => ActivityPub::CONTEXT, 'id' => $url, 'type' => 'Tombstone'];
+				if (!$failed && ($curlResult->getReturnCode() == 410)) {
+					$data = ['@context' => ActivityPub::CONTEXT, 'id' => $url, 'type' => 'Tombstone'];
+				}
+			} catch (\Throwable $th) {
```

Comment:
Why do you use `\Throwable` and not `\Exception` ?
For me, throwable is a little bit too generic, but it's just a feeling about it ..

Reply:
This is just what my editor is returning as template. In fact I don't know the difference between the different variants.

Reply:
Starting from PHP 7, PHP has thrown PHP errors as exceptions than can be caught in try catch blocks. It created a new ˋError` inheritance tree parallel to the existing `Exception` inheritance tree, which `Throwable` became the common parent class.